tests/utils/streams/test_kafka.py

```diff
@@ -17,268 +18,147 @@
-        [NewTopic(name, num_partitions=1, replication_factor=1)]
-    ).items()
-    assert key == name
-    assert future.result() is None
-    try:
-        yield Topic(name)
-    finally:
-        [[key, future]] = client.delete_topics([name]).items()
-        assert key == name
-        assert future.result() is None
+    def decode(self, value: KafkaPayload) -> int:
+        return int(value.value.decode("utf-8"))
 
 
-def build_producer() -> KafkaProducer[KafkaPayload]:
-    codec: PassthroughCodec[KafkaPayload] = PassthroughCodec()
-    return KafkaProducer(configuration, codec=codec)
+class KafkaStreamsTestCase(StreamsTestMixin, TestCase):
 
+    configuration = {"bootstrap.servers": "127.0.0.1"}
+    codec = TestCodec()
 
-def test_consumer_backend(topic: Topic) -> None:
-    def build_consumer() -> Consumer[KafkaPayload]:
-        codec: PassthroughCodec[KafkaPayload] = PassthroughCodec()
+    @contextlib.contextmanager
+    def get_topic(self) -> Iterator[Topic]:
+        name = f"test-{uuid.uuid1().hex}"
+        client = AdminClient(self.configuration)
+        [[key, future]] = client.create_topics(
+            [NewTopic(name, num_partitions=1, replication_factor=1)]
+        ).items()
+        assert key == name
+        assert future.result() is None
+        try:
+            yield Topic(name)
+        finally:
+            [[key, future]] = client.delete_topics([name]).items()
+            assert key == name
+            assert future.result() is None
+
+    def get_consumer(
+        self, group: str, auto_offset_reset: str = "earliest"
+    ) -> KafkaConsumer[int]:
         return KafkaConsumer(
             {
-                **configuration,
-                "auto.offset.reset": "earliest",
+                **self.configuration,
+                "auto.offset.reset": auto_offset_reset,
                 "enable.auto.commit": "false",
                 "enable.auto.offset.store": "false",
                 "enable.partition.eof": "true",
-                "group.id": "test",
+                "group.id": group,
                 "session.timeout.ms": 10000,
             },
-            codec=codec,
+            self.codec,
         )
 
-    with closing(build_producer()) as producer:
-        value = uuid.uuid1().hex.encode("utf-8")
-        assert (
-            wait(
-                [producer.produce(topic, KafkaPayload(None, value)) for i in range(2)],
-                timeout=5.0,
-            ).not_done
-            == set()
+    def get_producer(self) -> KafkaProducer[int]:
+        return KafkaProducer(self.configuration, self.codec)
+
+    def test_auto_offset_reset_earliest(self) -> None:
+        with self.get_topic() as topic:
+            with closing(self.get_producer()) as producer:
+                producer.produce(topic, 0).result(5.0)
+
+            with closing(
+                self.get_consumer("test", auto_offset_reset="earliest")
+            ) as consumer:
+                consumer.subscribe([topic])
```

Comment:
I feel like there must be a cleaner abstraction here for splitting out this setup code on a per-method basis, but I can't think of what it would be without parameterizing the assertion itself as well, which seems weird here.